### PR TITLE
Moddle Login with Dio

### PR DESCRIPTION
**Pull request by kurwjan**
+ Login with Dio
+ Update WebView Package
+ Display error on top of WebView
   + Should support every platform than displaying the error under the WebView
+ Display no internet warning
+ Switch  `LessonsFile` with `FileInfo` and put it in `file_operations.dart` instead of being a part of Lessons.
+ Display file modal when downloading smth.
   + Add a bit padding to it
+ Copy error url when in error screen

### CODE REVIEW

**Review by Rajala1404 — CHANGES_REQUESTED** [review]
Internet warning does not get shown in File Storage.
And for some reason everything there is German (but i fix that in my UX PR)
image

**Comment by kurwjan** [comment]
> Internet warning does not get shown in File Storage.
> And for some reason everything there is German (but i fix that in my UX PR)
> image
> 

No internet warning in webview, not file storage.

**Review by Rajala1404 — DISMISSED** [review]
Ah ok

File: app/lib/view/moodle.dart [hunk 1]
```diff
@@ -16,46 +22,152 @@ class MoodleWebView extends StatefulWidget {
...
+        value: cookies[i].value,
+        path: cookies[i].path!,
+        domain: cookies[i].domain,
+        isHttpOnly: cookies[i].httpOnly,
+        isSecure: cookies[i].secure,
+      );
+    }
+  }
+
+  Future<void> getCookies() async {
```
Comment (alessioC42) [inline app/lib/view/moodle.dart:62]: I see pain.

**Review by alessioC42 — APPROVED** [review]

**Pull request merged**